src/contracts/composers/FraxtalLZCurveComposer.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import { IOAppCore } from "@layerzerolabs/oapp-evm/contracts/oapp/interfaces/IOAppCore.sol";
+import { IOAppComposer } from "@layerzerolabs/oapp-evm/contracts/oapp/interfaces/IOAppComposer.sol";
+import { OFTComposeMsgCodec } from "@layerzerolabs/oft-evm/contracts/libs/OFTComposeMsgCodec.sol";
+
+import { OptionsBuilder } from "@fraxfinance/layerzero-v2-upgradeable/oapp/contracts/oapp/libs/OptionsBuilder.sol";
+import { SendParam, MessagingFee, IOFT } from "@fraxfinance/layerzero-v2-upgradeable/oapp/contracts/oft/interfaces/IOFT.sol";
+
+import { FraxtalStorage, FraxtalL2 } from "src/contracts/FraxtalStorage.sol";
+
+interface ICurve {
+    function exchange(int128 i, int128 j, uint256 _dx, uint256 _min_dy) external returns (uint256);
+    function get_balances() external view returns (uint256[] memory);
+}
+
+interface IWETH {
+    function withdraw(uint256 wad) external;
+    function deposit() external payable;
+}
+
+contract FraxtalLZCurveComposer is IOAppComposer, Initializable, FraxtalStorage {
+    error FailedEthTransfer();
+
+    // keccak256(abi.encode(uint256(keccak256("frax.storage.FraxtalLZCurveComposer")) - 1))
+    bytes32 private constant FraxtalLZCurveComposerStorageLocation =
+        0x907a4b366d5e5f18e135695c3ea756f3cd95a707ab3a6c8d7d99d1e1b612ce1e;
+    struct FraxtalLZCurveComposerStorage {
+        address endpoint;
+    }
+
+    function _getFraxtalLZCurveComposerStorage() private pure returns (FraxtalLZCurveComposerStorage storage $) {
+        assembly {
+            $.slot := FraxtalLZCurveComposerStorageLocation
+        }
+    }
+    constructor() {
+        _disableInitializers();
```

Comment:
Discussion: I think we may be over fitting for upgradability on these contracts. 

We have a setter for this variable w/n `src/contracts/amos/FraxtalLZCurveAMO.sol` 

I am curious as to the motivation behind the choice of having proxies and then settable variables referencing said proxies.

Reply:
You may be right that some of these contracts don't need to be upgradeable.

Part of the reason behind proxies is because of the Frax Vision 2030: I don't know if contract addresses or wrap-ability will change of tokens so I'd like to keep the contracts flexible in case there's a difference in contract functionality.  Ultimately, I want to reduce the need to re-deploy and making the contracts slightly more complicated is worth it IMO.  Do you think there would be a security concern from users in using non-immutable contracts here?

The setters within `FraxtalLZCurveAMO` are needed because of the operations flow: the Ethereum contracts need the Fraxtal AMO address and the Fraxtal AMO needs the Ethereum addresses.  Which means we have to call setters on one side to fully plug together the contracts (unless we went with create2 pre-determined addresses but I don't believe that's necessary here).